internal/provider/resource_password.go

```diff
@@ -535,8 +535,21 @@ func upgradePasswordStateV2toV3(ctx context.Context, req resource.UpgradeStateRe
 	resp.Diagnostics.Append(resp.State.Set(ctx, passwordDataV3)...)
 }
 
+// generateHash truncates strings that are longer than 72 bytes in
+// order to avoid the error returned from bcrypt.GenerateFromPassword
+// in versions v0.5.0 and above (refer to
+// https://pkg.go.dev/golang.org/x/crypto/bcrypt?tab=versions).
+// This is to preserve backwards compatibility and will be removed in
+// a subsequent version of the provider.
```

Comment:
Is the plan to validate password length to a maximum of 72 bytes in a future version or should this comment be removed?

Reply:
I was thinking in the next major version release we could generate an error if the password length exceeds 72 bytes. If that sounds ok, I'll create a follow-up issue and link it in the code comment.

Reply:
Please feel free to raise a proposal, however I'm a little worried that may be an unpopular proposal just to appease a specific hashing algorithm behavior that not all affected practitioners (anyone generating long passwords) may need from the resource. I think noting the limitation in that hash description will either be good enough for the purposes of this provider or another proposal could be made to move that hashing algorithm to another provider (potentially in a provider-defined function once available) and drop the "problematic" attribute as out of scope for this provider.

Reply:
I removed the statement of removal from the comment and we can determine in a later issue if that's something we want to do ⚡